src/Filters/stream_filt.jl

```diff
@@ -562,18 +630,42 @@ end
 # Stateless filt implementations
 #
 
-# Decimation, interpolation, and rational resampling.
-function Base.filt(h::Vector, x::Vector, ratio::Rational)
+# Single-rate, decimation, interpolation, and rational resampling.
+function Base.filt(h::Vector, x::AbstractVector, ratio::Rational=1//1)
     self = FIRFilter(h, ratio)
     filt(self, x)
 end
 
 # Arbitrary resampling with polyphase interpolation and two neighbor lnear interpolation.
-function Base.filt(h::Vector, x::Vector, rate::FloatingPoint, Nϕ::Integer=32)
+function Base.filt(h::Vector, x::AbstractVector, rate::FloatingPoint, Nϕ::Integer=32)
     self = FIRFilter(h, rate, Nϕ)
     filt(self, x)
 end
 
+function resample(x::AbstractVector, rate::Real, h::Vector)
+    self = FIRFilter(h, rate)
+
+    # Get delay, in # of samples at the output rate, caused by filtering processes
+    τ = timedelay(self)
+
+    # Use setphase! to
+    #   a) adjust the input samples to skip over before producing and output (integer part of τ)
+    #   b) set the ϕ index of the PFB (fractional part of τ)
+    setphase!(self, τ)
+
+    # Calculate the number of 0's required so that w
+    outLen      = ceil(Int, length(x)*rate)
+    reqInlen    = inputlength(self, outLen)
+    reqZerosLen = reqInlen - length(x)
+    xPadded     = [x, zeros(eltype(x), reqZerosLen)]
```

Comment:
On 0.4 this should be `[x; zeros(eltype(x), reqZerosLen)]` to avoid a deprecation warning, but perhaps you could avoid copying `x` with something like:

```julia
out = Array(Base.promote_eltype(x, h), outLen)
n = filt!(out, self, x)
filt!(sub(out, n+1:outLen), self, zeros(eltype(x), reqZerosLen))
out
```